src/components/Checkbox/Checkbox.tsx

```diff
@@ -105,23 +108,62 @@ const CheckboxComponent: ForwardRefRenderFunction<HTMLInputElement, CheckboxProp
             isDisabled: disabled,
             isRequired: required,
             isIndeterminate: state === CheckboxState.Mixed,
-            'aria-label': ariaLabel || label,
+            'aria-label': ariaLabel,
             value,
         },
         toggleState,
         inputRef,
     );
 
+    const emphasisCheckedClassesMap: Record<CheckboxEmphasis, string> = {
```

Comment:
All those tailwind classes mapping can be moved outside of the component so we can keep it as clean as possible.